circle.yml

```diff
@@ -4,96 +4,143 @@
 #   circle.yml in the kbfs repository!
 #
 # -----------------------------------------------------------------------------
-machine:
-  xcode:
-    version: 8.3.3
-  node:
-    version: 8
-  java:
-    version: oraclejdk8
-  environment:
-    GOPATH: $HOME/go-path
-    PATH: $GOROOT/bin:$PATH
-    # Limit memory usage of gradle
-    _JAVA_OPTIONS: "-Xms512m -Xmx2048m"
-    GRADLE_OPTS: '-Dorg.gradle.jvmargs="-Xmx2048m -XX:+HeapDumpOnOutOfMemoryError"'
-    REACT_NATIVE_MAX_WORKERS: 1
-    # Some node dependencies break without this
-    CXX: g++-4.8
-    ANDROID_NDK_HOME: ${ANDROID_NDK}
-general:
-  build_dir: shared
-dependencies:
-  pre:
-    # - nvm ls-remote
-    - sudo apt-key adv --fetch-keys http://dl.yarnpkg.com/debian/pubkey.gpg
-    - echo "deb http://dl.yarnpkg.com/debian/ stable main" | sudo tee /etc/apt/sources.list.d/yarn.list
-    - sudo apt-get update -qq
-    # - sudo apt-cache show yarn
-    - sudo apt-get install -y -qq yarn=1.0.2-1
-    - sudo rm -rf /usr/local/go* # Remove old go installation
-    - mkdir -p ~/downloads
-    - mkdir -p ~/go-path
-    - ls "$HOME/downloads"
-    - ls "$ANDROID_HOME"
-    - if [[ ! -e "$HOME/downloads/go.19.tar.gz" ]]; then wget -nv https://storage.googleapis.com/golang/go1.9.linux-amd64.tar.gz -O $HOME/downloads/go.19.tar.gz ; fi
-    - cd $HOME && tar -xzvf downloads/go.19.tar.gz
-    - sudo mv $HOME/go /usr/local
-    - if ! $(grep -q "Revision=24.4.1" $ANDROID_HOME/tools/source.properties); then echo y | android update sdk -u -a -t "tools"; fi
-    - if [ ! -e $ANDROID_HOME/build-tools/23.0.2 ]; then echo y | android update sdk -u -a -t "build-tools-23.0.2"; fi
-     # Android Support Repository, revision 35 / Local Maven repository for Support Libraries
-    - if [ ! -d "/usr/local/android-sdk-linux/extras/android/m2repository/com/android/support/design/24.1.0" ]; then echo y | android update sdk --no-ui --all --filter "extra-android-m2repository"; fi
+version: 2
+jobs:
+  build:
+    working_directory: ~/client
```

Comment:
Changed from `~/client/shared`, as checkout step would balk at checking out into `~/client` (since it would be non-empty)